From 0328e3d32c6915650b14dd40fcd7598a420b1364 Mon Sep 17 00:00:00 2001
From: Konstantin Khlebnikov <khlebnikov@openvz.org>
Date: Tue, 24 Feb 2009 16:47:23 +0300
Subject: [PATCH] pidns: update leader_pid at pidns attach

after commit fea9d17 it_real_fn send SIGALRM to task->signal->leader_pid
(used for sys_alarm(...) and sys_setitimer(ITIMER_REAL,...))

Thus, __pid_ns_attach_task hack-n-dirty cross pid-ns task movement must
update this pid too

http://bugzilla.openvz.org/show_bug.cgi?id=1160
127384

Signed-off-by: Konstantin Khlebnikov <khlebnikov@openvz.org>
Signed-off-by: Pavel Emelyanov <xemul@openvz.org>
---
 kernel/pid_namespace.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/kernel/pid_namespace.c b/kernel/pid_namespace.c
index c478b80..1445b22 100644
--- a/kernel/pid_namespace.c
+++ b/kernel/pid_namespace.c
@@ -209,6 +209,7 @@ static int __pid_ns_attach_task(struct pid_namespace *ns,
 	set_task_session(tsk, pid_nr(pid));
 	reattach_pid(tsk, PIDTYPE_PGID, pid);
 	tsk->signal->__pgrp = pid_nr(pid);
+	tsk->signal->leader_pid = pid;
 	current->signal->tty_old_pgrp = NULL;
 
 	reattach_pid(tsk, PIDTYPE_PID, pid);
-- 
1.6.0.6

